test/test_node.go

```diff
@@ -266,46 +268,43 @@ func isReachable(wg *sync.WaitGroup, t *testing.T, host ContainerPort, ch chan<-
 	ch <- connectionAttempt(t, host)
 }
 
-func (hosts Hosts) WaitForAllToStart(t *testing.T, timeout int) {
+func (hosts Hosts) WaitForAllToStart(t *testing.T, timeout int) error {
 	if len(hosts) == 0 {
-		return
+		return nil
 	}
 	for seconds := 1; seconds <= timeout; seconds++ {
 		var wg sync.WaitGroup
 
 		results := make(chan bool, len(hosts))
-		for i := 0; i < len(hosts); i++ {
-			host := hosts[i]
-			wg.Add(1)
+		wg.Add(len(hosts))
+		for _, host := range hosts {
 			go isReachable(&wg, t, host, results)
 		}
+		wg.Wait()
 
-		go func() {
-			wg.Wait()
-			close(results)
-		}()
+		close(results)
 
 		foundUnreachable := false
 
 		for reachable := range results {
 			if !reachable {
-				t.Logf("A signer node is not reachable")
+				t.Logf("A host is not reachable")
 				foundUnreachable = true
 				break
 			}
 		}
 		if foundUnreachable {
 			continue
 		}
-		t.Logf("All signers are reachable after %d seconds", seconds)
-		return
+		t.Logf("All hosts are reachable after %d seconds", seconds)
+		return nil
 	}
-	t.Logf("Timed out after %d seconds waiting for signers", timeout)
+	return fmt.Errorf("timed out after %d seconds waiting for hosts", timeout)
 }
 
 // Name is the hostname of the test node container
 func (tn *TestNode) Name() string {
-	return fmt.Sprintf("node-%d-%s", tn.Index, tn.t.Name())
+	return fmt.Sprintf("val-%d-node-%d-%s", tn.ValidatorIndex, tn.Index, tn.t.Name())
```

Comment:
Using `(*testing.T).Name()` can contain a `/` if using subtests. This isn't the only piece that I think will break if trying to use subtests, but it seemed worth mentioning. But it is fine to build an integration-style test framework that does not cooperate with subtests IMO.

Reply:
Sounds good, we can cross that bridge when/if we get there.